core/lib/types/src/commitment/mod.rs

```diff
@@ -350,6 +365,23 @@ impl L1BatchAuxiliaryOutput {
                 let state_diffs_hash = H256::from(keccak256(&(state_diffs_packed)));
                 let state_diffs_compressed = compress_state_diffs(state_diffs);
 
+                let blob1_linear_hash = system_logs
+                    .iter()
+                    .find_map(|log| {
+                        (log.0.sender == PUBDATA_CHUNK_PUBLISHER_ADDRESS
+                            && log.0.key == H256::from_low_u64_be(BLOB1_LINEAR_HASH_KEY as u64))
```

Comment:
Note, that for the 1.4.1 version the hash key `7` used the key for the upgrade tx hash, in 1.4.2 it was just shifted:
<img width="576" alt="image" src="https://github.com/matter-labs/zksync-era/assets/52464764/e2eb6894-2a62-454f-8f03-053c1c2d3267">

In other words, the code here will not work fine for the batch with the 1.4.1 upgrade (may be a problem for EN) or the batch where we had to do some emergency upgrade and unlike 1.4.2 we did not change the bootloader

Reply:
Nice catch! I will fix it by checking if version is post 1.4.2 before reading blob hashes from system logs